src/foam/dao/AbstractJDAO.java

```diff
@@ -283,49 +275,31 @@ public void removeAll_(final X x, long skip, final long limit, Comparator order,
     getDelegate().removeAll_(x, skip, limit, order, predicate);
   }
 
-  protected FObject difference(FObject o, FObject n) {
-    FObject diff = o.hardDiff(n);
-    // no difference, then return null
-    if ( diff == null ) return null;
-    // get the PropertyInfo for the id
-    PropertyInfo idInfo = (PropertyInfo) getOf().getAxiomByName("id");
-    // set id property to new instance
-    idInfo.set(diff, idInfo.get(o));
-    return diff;
-  }
-
-  protected FObject mergeChange(FObject o, FObject c) {
-    //if no change to merge, return FObject;
+  protected FObject mergeFObject(FObject o, FObject c) {
     if ( c == null ) return o;
-    //merge change
-    return maybeMerge(o, c);
-  }
-
-  protected FObject maybeMerge(FObject o, FObject c) {
-    if ( o == null ) return o = c;
 
     //get PropertyInfos
     List list = o.getClassInfo().getAxiomsByClass(PropertyInfo.class);
     Iterator e = list.iterator();
 
-    while ( e.hasNext() ) {
+    while( e.hasNext() ) {
       PropertyInfo prop = (PropertyInfo) e.next();
-      if ( prop instanceof AbstractFObjectPropertyInfo ) {
-        //do nested merge
-        //check if change
-        if ( ! prop.isSet(c) ) continue;
-        maybeMerge((FObject) prop.get(o), (FObject) prop.get(c));
-      } else {
-        //check if change
-        if ( ! prop.isSet(c) ) continue;
-        //set new value
-        prop.set(o, prop.get(c));
-      }
+      mergeProperty(o, c, prop);
     }
-
     return o;
   }
 
+  protected void mergeProperty(FObject o, FObject c, PropertyInfo prop) {
+    if ( ! prop.isSet(c) ) return;
+    if ( prop instanceof AbstractFObjectPropertyInfo ) {
```

Comment:
Is this block needed? We don't merge sub-objects do we? Don't we just send the whole object if it changed?